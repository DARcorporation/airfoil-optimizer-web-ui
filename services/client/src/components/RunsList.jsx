import React from 'react';

const RunsList = (props) => {
  return (
    <div>
      <h2 className="title is-2">Submitted Runs:</h2>
      {
        props.runs.map((run) => {
          return (
            <h4
              key={run.id}
              className="box title is-4"
              style={
                {
                  backgroundColor: run.status === 0 ? 'lightblue' : (
                    run.status === 1 ? 'orange' : (
                      run.status === 2 ? 'lightgreen' : 'red'
                    )
                  )
                }
              }
            >{
              run.status === 0 ? "In Queue" : (
                run.status === 1 ? "Running..." : (
                  run.status === 2 ? "Completed" : "Failed"
                )
              )
            }
            </h4>
          )
        })
      }
    </div>
  )
};

export default RunsList;