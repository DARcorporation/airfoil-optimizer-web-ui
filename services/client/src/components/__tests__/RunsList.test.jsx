import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import RunsList from "../RunsList";

const runs = [
  {
    'id': 1,
    'cl': 1.0,
    'n_c': 3,
    'n_t': 3,
    'b_c': 8,
    'b_t': 8,
    'b_te': 8,
    'gen': 100
  },
  {
    'id': 2,
    'cl': 0.5,
    'n_c': 6,
    'n_t': 6,
    'b_c': 16,
    'b_t': 16,
    'b_te': 16,
    'gen': 300
  },
];

test('RunsList renders properly', () => {
  const wrapper = shallow(<RunsList runs={runs}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toStrictEqual([1, ", ", 3, ", ", 3, ", ", 8, ", ", 8, ", ", 8, ", ", 100])
});

test('RunsList renders a snapshot properly', () => {
  const tree = renderer.create(<RunsList runs={runs}/>).toJSON();
  expect(tree).toMatchSnapshot();
});