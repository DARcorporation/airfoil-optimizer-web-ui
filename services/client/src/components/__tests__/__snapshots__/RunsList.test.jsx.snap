// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunsList renders a snapshot properly 1`] = `
<div>
  <h2
    className="title is-2"
  >
    Submitted Runs:
  </h2>
  <h4
    className="box title is-4"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    Failed
  </h4>
  <h4
    className="box title is-4"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    Failed
  </h4>
</div>
`;
